src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -242,7 +242,8 @@ private IEnumerable<MgmtClientOperation> EnsureAllOperations()
                     if (IsListOperation(operation, operationSet))
                     {
                         // if this operation is a collection operation, it should be the parent of its corresponding resource request path
-                        var diff = requestTrimmedPath.TrimAncestorFrom(resourceTrimmedPath);
+                        // In the case of /{scope}/providers/Microsoft.Resources/links being a list method for /{linkId}, we need to reverse the trim.
+                        var diff = requestTrimmedPath.SerializedPath.Length < resourceTrimmedPath.SerializedPath.Length ? requestTrimmedPath.TrimPrefixPathFrom(resourceTrimmedPath) : requestTrimmedPath.TrimPrefixPathFrom(requestTrimmedPath);
```

Comment:
In our latest changes, we change the implementation around here, you might need to have a look if this change is still needed in the latest code

Reply:
The change is no longer needed and reverted.